core/src/main/java/com/scalar/db/common/CoreError.java

```diff
@@ -931,6 +931,72 @@ public enum CoreError implements ScalarDbError {
       "Conditions on indexed columns in cross-partition scan operations are not allowed in the SERIALIZABLE isolation level",
       "",
       ""),
+  VIRTUAL_TABLE_NOT_SUPPORTED_IN_STORAGE(
+      Category.USER_ERROR,
+      "0263",
+      "To support virtual tables, the atomicity unit of the storage must be at least at the namespace level. Storage: %s; Atomicity unit: %s",
+      "",
+      ""),
+  VIRTUAL_TABLE_SOURCE_TABLES_OUTSIDE_OF_ATOMICITY_UNIT(
+      Category.USER_ERROR,
+      "0264",
+      "The source tables must reside within the atomicity unit of the storage. Storage: %s; Atomicity unit: %s; Left source table: %s; Right source table: %s",
+      "",
+      ""),
+  DYNAMO_VIRTUAL_TABLE_NOT_SUPPORTED(
```

Comment:
This is just an idea, but how about making this more generic? It might be helpful if the error code is reused for other future storages like this:
```
 VIRTUAL_TABLE_NOT_SUPPORTED(
      Category.USER_ERROR,
      "0265",
      "The virtual table functionality is not supported in %s",
      "",
      ""),
```